plugins/otter-pro/inc/plugins/class-block-conditions.php

```diff
@@ -566,6 +596,122 @@ public function has_products( $products ) {
 		return $bought;
 	}
 
+	/**
+	 * Check based on WooCommerce product category.
+	 *
+	 * @param array $categories IDs of Categories.
+	 *
+	 * @since  2.7.0
+	 * @access public
+	 */
+	public function has_product_category( $categories ) {
+		$in_category = false;
+
+		global $product;
+
+		if ( ! $product ) {
+			return false;
+		}
+
+		$terms = get_the_terms( $product->get_id(), 'product_cat' );
+
+		if ( gettype( $terms ) !== 'array' ) {
+			return false;
+		}
+
+		foreach ( $terms as $term ) {
+			if ( in_array( $term->term_id, $categories, true ) ) {
+				$in_category = true;
+				break;
+			}
+		}
+
+		return $in_category;
+	}
+
+	/**
+	 * Check based on WooCommerce product tags.
+	 * 
+	 * @param array $tags IDs of Tags.
+	 * 
+	 * @since  2.7.0
+	 * @access public
+	 */
+	public function has_product_tag( $tags ) {
+		$in_tag = false;
+
+		global $product;
+
+		if ( ! $product ) {
+			return false;
+		}
+
+		$terms = get_the_terms( $product->get_id(), 'product_tag' );
+
+		if ( gettype( $terms ) !== 'array' ) {
+			return false;
+		}
+
+		foreach ( $terms as $term ) {
+			if ( in_array( $term->term_id, $tags, true ) ) {
+				$in_tag = true;
+				break;
+			}
+		}
+
+		return $in_tag;
+	}
+
+	/**
+	 * Check based on WooCommerce product attribute.
+	 *
+	 * @param array $condition Condition.
+	 *
+	 * @since  2.7.0
+	 * @access public
+	 */
+	public function has_product_attribute( $condition ) {
+		$attribute     = $condition['attribute'];
```

Comment:
This is where I'd add the `isset` check. You can just return `false` if there's nothing.

Reply:
As mentioned before, we don't want to check the condition if the attribute doesn't exist, so instead of returning true/false, we just want to skip the condition check altogether.